pkg/util/xds/metric_sanitizer.go

```diff
@@ -0,0 +1,18 @@
+package xds
+
+import (
+	"regexp"
+	"strings"
+)
+
+var (
+	whitespaces  = regexp.MustCompile(`\s+`)
+	illegalChars = regexp.MustCompile(`[^a-zA-Z_\-0-9]`)
+)
+
+func SanitizeMetric(metric string) string {
+	result := whitespaces.ReplaceAllString(metric, "_")
+	result = strings.ReplaceAll(result, "/", "_")
+	result = illegalChars.ReplaceAllString(result, "_")
```

Comment:
Why do we need 3 steps ?

Wouldn't `return illegalChars.ReplaceAllString(metric, "_")` be enough ?